xdem/spatial_tools.py

```diff
@@ -396,3 +396,42 @@ def hillshade(dem: Union[np.ndarray, np.ma.masked_array], resolution: Union[floa
     # Return the hillshade, scaled to uint8 ranges.
     # The output is scaled by "(x + 0.6) / 1.84" to make it more similar to GDAL.
     return np.clip(255 * (shaded + 0.6) / 1.84, 0, 255).astype("float32")
+
+
+def subsample_raster(array: Union[np.ndarray, np.ma.masked_array], subsample: Union[float, int],
```

Comment:
The typing is still a bit off in this.. It says that it returns an ndarray, but it may also return a tuple of ndarrays. I suggest that if `return_indices=True`, it only returns the indices, not the array (which is easily created by the user). That way, the return type is consistent.

Reply:
Done in 90bd065